tldextract/remote.py

```diff
@@ -49,13 +57,11 @@ def looks_like_ip(maybe_ip: str) -> bool:
     if not maybe_ip[0].isdigit():
         return False
 
-    try:
-        socket.inet_aton(maybe_ip)
-        return True
-    except (AttributeError, UnicodeError):
-        if IP_RE.match(maybe_ip):
+    if inet_pton is not None:
+        try:
```

Comment:
If I comment out this block, i.e. `inet_pton` is always unavailable, the test suite still passes. Can you add an assert demonstrating the use of `inet_pton`?

```diff
diff --git a/tldextract/remote.py b/tldextract/remote.py
index 1c6cad2..60ca31a 100644
--- a/tldextract/remote.py
+++ b/tldextract/remote.py
@@ -3,14 +3,15 @@
 from __future__ import annotations
 
 import re
-from collections.abc import Callable
+
+# from collections.abc import Callable
 from urllib.parse import scheme_chars
 
-inet_pton: Callable[[int, str], bytes] | None
-try:
-    from socket import AF_INET, inet_pton  # Availability: Unix, Windows.
-except ImportError:
-    inet_pton = None
+# inet_pton: Callable[[int, str], bytes] | None
+# try:
+#     from socket import AF_INET, inet_pton  # Availability: Unix, Windows.
+# except ImportError:
+#     inet_pton = None
 
 IP_RE = re.compile(
     r"^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.)"
@@ -57,11 +58,11 @@ def looks_like_ip(maybe_ip: str) -> bool:
     if not maybe_ip[0].isdigit():
         return False
 
-    if inet_pton is not None:
-        try:
-            inet_pton(AF_INET, maybe_ip)
-            return True
-        except OSError:
-            return False
+    # if inet_pton is not None:
+    #     try:
+    #         inet_pton(AF_INET, maybe_ip)
+    #         return True
+    #     except OSError:
+    #         return False
 
     return IP_RE.match(maybe_ip) is not None
```

Reply:
Do you means something like `assert inet_pton is None`?

Reply:
Oh, that's an improvement!

What I originally meant was testing `inet_aton` vs `inet_pton`. But that's already covered by an assert you added later in this file. With `inet_aton`, the assert fails, with `inet_pton` it passes.

```
    assert_extract(
        "http://127.0.0/foo/bar", ("", "127.0", "0", ""), expected_ip_data=""
    )
```

I checked out master with your new tests and figured that out. So, this PR all good!